### PR TITLE
Modify makefile and publish workflow

### PR DESCRIPTION
**Pull request by AngMits**
## 📝 Description

- Add the ability to publish to PyPI via poetry
- Add the respective secret (pending deposition) for PyPI
- Rename workflow for better accuracy

Closes issue #101

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,7 +42,12 @@ clean:
 build:
 	@poetry build
 
-publish:
+publish-to-jfrog:
 	@poetry publish --repository jfrog --username $(ARTIFACTORY_USERNAME) --password $(ARTIFACTORY_API_KEY)
 
+publish-to-pypi:
+	@poetry config pypi-token.pypi $(PYPI_TOKEN) && @poetry publish
```
Comment (antonakospanos) [inline Makefile:49]: We can skip the `config` command as follows:

```suggestion
	@poetry publish --username __token__ --password $(PYPI_TOKEN)
```

**Review by antonakospanos — APPROVED** [review]
Verified publishing our latest release at https://pypi.org/project/g3/

**Comment by AngMits** [comment]
@antonakospanos The latest PR update follows a slightly different path from the proposed one. The variables are constructed for use inside the action, and not independently. Do you think it would be better to provide an alternative, in case we need to manually publish a version? Do keep in mind that the workflow can be run on demand as well though.

**Comment by antonakospanos** [comment]
> @antonakospanos The latest PR update follows a slightly different path from the proposed one. The variables are constructed for use inside the action, and not independently. Do you think it would be better to provide an alternative, in case we need to manually publish a version? Do keep in mind that the workflow can be run on demand as well though.

[Verified](https://pypi.org/project/g3/). Even better approach relying on `POETRY_PYPI_TOKEN_PYPI` env var 👏

**Pull request merged**